servicetalk-http-api/src/main/java/io/servicetalk/http/api/RedirectConfiguration.java

```diff
@@ -0,0 +1,165 @@
+     *
+     * @param methods {@link HttpRequestMethod}s that are allowed to follow redirects
+     * @return {@code this}
+     */
+    RedirectConfiguration allowedMethods(HttpRequestMethod... methods);
+
+    /**
+     * Defines an additional check to decide if the redirect should be performed or not.
+     * <p>
+     * This predicate runs as the last check, after validation of the {@link HttpResponseStatus},
+     * {@link #maxRedirects(int)}, {@link #allowedMethods(HttpRequestMethod...)}, and
+     * {@link HttpHeaderNames#LOCATION Location} header.
+     *
+     * @param shouldRedirect {@link BiPredicate} to decide if the request should follow redirect or not given the
+     * original request and received response meta-data
+     * @return {@code this}
+     */
+    RedirectConfiguration shouldRedirect(BiPredicate<HttpRequestMetaData, HttpResponseMetaData> shouldRedirect);
+
+    /**
+     * Enforces change from {@link HttpRequestMethod#POST POST} to {@link HttpRequestMethod#GET GET} for subsequent
+     * requests for {@link HttpResponseStatus#MOVED_PERMANENTLY 301 (Moved Permanently)} and
+     * {@link HttpResponseStatus#FOUND 302 (Found)} status codes.
+     * <p>
+     * For historical reasons, RFC7231 sections
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7231#section-6.4.2">6.4.2</a> and
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7231#section-6.4.3">6.4.3</a> allow user agents to change the
+     * request method from {@link HttpRequestMethod#POST POST} to {@link HttpRequestMethod#GET GET} for the subsequent
+     * request. If this behavior is undesired, this option can be turned off or
+     * {@link HttpResponseStatus#TEMPORARY_REDIRECT 307 (Temporary Redirect)} and
+     * {@link HttpResponseStatus#PERMANENT_REDIRECT 308 (Permanent Redirect)} status codes can be used instead.
+     *
+     * @param changePostToGet if {@code true}, request method will change from {@link HttpRequestMethod#POST POST}
+     * to {@link HttpRequestMethod#GET GET} while following
+     * {@link HttpResponseStatus#MOVED_PERMANENTLY 301 (Moved Permanently)} or
+     * {@link HttpResponseStatus#FOUND 302 (Found)} redirect status codes
+     * @return {@code this}
+     */
+    RedirectConfiguration changePostToGet(boolean changePostToGet);
+
+    /**
+     * Headers that have to be copied from the original request on each non-relative redirect.
+     * <p>
+     * Note: for security reasons, redirection should not automatically copy any headers from the original request when
+     * it performs a non-relative redirect.
+     *
+     * @param headerNames Names of headers that have to be copied on each non-relative redirect
+     * @return {@code this}
+     * @see #allowNonRelativeRedirects(boolean)
+     * @see #redirectPayloadBody(boolean)
+     * @see #trailersToRedirect(CharSequence...)
+     * @see #prepareRequest(TriConsumer)
+     */
+    RedirectConfiguration headersToRedirect(CharSequence... headerNames);
```

Comment:
If users need more flexibility and control, they can use `prepareRequest` that gives access to the original request and redirect request

Reply:
WDYT, should we also add `redirectAllHeaders/redirectAllTrailers` to make it even easier or it may open extra risk (users won't think what headers are sensitive)? They can also do:
```java
.prepareRequest((prevRequest, redirectResponse, redirectedRequest) -> {
    redirectedRequest.setHeaders(prevRequest.headers());
}))
```

Reply:
discussed offline ... I like the `DefaultRedirectRequestTransformer` approach to consolidate responsibility and give more flexibility🏅